src/sentry/eventstream/snuba.py

```diff
@@ -97,6 +97,7 @@ def insert(
         primary_hash,
         received_timestamp,  # type: float
         skip_consume=False,
+        assign_partitions_randomly=False,
```

Comment:
I don't think the public interface of the insert of `SnubaProtocolEventStream` should know about assigning partitions randomly or not. `SnubaProtocolEventStream` is also used to send events to snuba via the web interface. The parameter doesn't make sense in that context. Is it possible to make this decision internally in the Kafka backend only without changing the public interface?

Reply:
Unfortunately, that's not possible, without violating a requirement set by Filippo that the `transaction_forwarder` header logic needs to be reused explicitly if we're tying group_id None-ness to whether an event is an error or transaction. Unless we want to split that `_is_transaction_event` method into a third module used by `SnubaProtocolEventStream` and `KafkaEventStream` which seems like it would be less clear to me.

Today, pre-merge, the application flow calls `KafkaEventStream#insert` which is a passthrough to `SnubaProtocolEventStream#insert` which constructs the final `_send` call (line 127) which is an abstracted call to produce either a kafka message or a HTTP post [for tests]. Since we don't want `SnubaProtocolEventStream` to set the header and the `_send` inside of `KafkaEventStream` doesn't have access to the raw event, we need the message type or the kill switch value to flow through the public method.

Would you prefer a more generic `message_options` hash/dataclass where `skip_semantic_partitioning` could be a key/property set to true or false?

Reply:
`message_options` should work. Or you could probably do kwargs. I will leave it to you to decide.

Reply:
did this with kwargs, seems like it's more of a standard Python idiom